### ISSUE TITLE
Warn when inputs are missing aria-labels

### ISSUE THREAD

**Issue by tjsavage** [PolymerLabs/polylint#14]

**Comment by garlicnation** [comment]
All aria labels:
```
aria-activedescendant
aria-atomic
aria-autocomplete
aria-busy (state)
aria-checked (state)
aria-controls
aria-describedby
aria-disabled (state)
aria-dropeffect
aria-expanded (state)
aria-flowto
aria-grabbed (state)
aria-haspopup
aria-hidden (state)
aria-invalid (state)
aria-label
aria-labelledby
aria-level
aria-live
aria-multiline
aria-multiselectable
aria-orientation
aria-owns
aria-posinset
aria-pressed (state)
aria-readonly
aria-relevant
aria-required
aria-selected (state)
aria-setsize
aria-sort
aria-valuemax
aria-valuemin
aria-valuenow
aria-valuetext
role
```

**Comment by alice** [comment]
@garlicnation Those are actually all the ARIA attributes, not just labels.

Detecting missing ARIA is kind of AI-hard, since it amounts to figuring out what the implicit semantics of the element are - and if we could do that programmatically, we probably wouldn't need ARIA :stuck_out_tongue:

**Comment by notwaldorf** [comment]
I think the best we can do is use the `a11ySuite` in the tests, which will determine at runtime if the right labels are applied. Could polylint maybe warn you if you've written a test that has a `test-fixture`, and there's no `a11ySuite` testing that fixture?

**Comment by azakus** [comment]
@alice I assume that accessibility dev tools evaluates labels with the algorithm you described?
Maybe it can be enough to use `a11ySuite` in conjunction with `polylint` to cover all those cases?

**Comment by alice** [comment]
@azakus It checks it for natively labelable elements and `<img>` elements, but not for all elements - which seems like a bug tbh.

You could use axs.properties.findTextAlternatives to assert that some text alternative exists in your code.
